contracts/payments/PaymentRouterERC20.sol

```diff
@@ -0,0 +1,296 @@
+    error IncorrectPaymentAmount();
+    /// @notice Error thrown when attempting to withdraw with no funds
+    error NoFundsToWithdraw();
+    /// @notice Error thrown when attempting to set an empty URI
+    error EmptyUri();
+    /// @notice Error thrown when token is not whitelisted
+    error TokenNotWhitelisted();
+    /// @notice Error thrown when token address is invalid
+    error InvalidTokenAddress();
+    /// @notice Error thrown when token is already whitelisted
+    error TokenAlreadyWhitelisted();
+    /// @notice Error thrown when token is not in whitelist
+    error TokenNotInWhitelist();
+
+    /**
+     * @notice Contract constructor
+     * @param _multiSigWallet Address of the multisig wallet
+     * @param managerRole Address to receive manager role
+     * @param adminRole Address to receive admin role
+     */
+    constructor(
+        address _multiSigWallet,
+        address managerRole,
+        address adminRole
+    ) {
+        if (_multiSigWallet == address(0)) revert InvalidMultiSigAddress();
+        multiSigWallet = _multiSigWallet;
+
+        _grantRole(DEFAULT_ADMIN_ROLE, adminRole);
+        _grantRole(ADMIN_ROLE, adminRole);
+        _grantRole(DEV_CONFIG_ROLE, msg.sender);
+        _grantRole(MANAGER_ROLE, managerRole);
+    }
+
+    /**
+     * @notice Modifier to check if a payment ID is valid
+     * @param id The payment ID to check
+     */
+    modifier validId(uint256 id) {
+        if (paymentConfigs[id].price == 0) revert InvalidPaymentId();
+        _;
+    }
+
+    /**
+     * @notice Adds a token to the whitelist
+     * @param token The token address to whitelist
+     */
+    function whitelistToken(address token) external onlyRole(MANAGER_ROLE) {
+        if (token == address(0)) revert InvalidTokenAddress();
+        if (whitelistedTokens[token]) revert TokenAlreadyWhitelisted();
+
+        whitelistedTokens[token] = true;
+        emit TokenWhitelisted(token);
+    }
+
+    /**
+     * @notice Removes a token from the whitelist
+     * @param token The token address to remove
+     */
+    function removeTokenFromWhitelist(
+        address token
+    ) external onlyRole(MANAGER_ROLE) {
+        if (!whitelistedTokens[token]) revert TokenNotInWhitelist();
+
+        whitelistedTokens[token] = false;
+        emit TokenRemovedFromWhitelist(token);
+    }
+
+    /**
+     * @notice Sets the payment configuration for a given ID
+     * @param id The payment ID
+     * @param price The price in token amount
+     * @param uri The metadata URI
+     * @param token The ERC20 token address for payment
+     */
+    function setPaymentConfig(
+        uint256 id,
+        uint256 price,
+        string calldata uri,
+        address token
+    ) external onlyRole(DEV_CONFIG_ROLE) {
+        if (price == 0) revert ZeroPrice();
+        if (bytes(uri).length == 0) revert EmptyUri();
+        if (!whitelistedTokens[token]) revert TokenNotWhitelisted();
+
+        paymentConfigs[id].paymentURI = uri;
```

Comment:
isPaused is false by default?

Reply:
yes, default solidity value is `false` into the setPaymentConfig for the struct